### PR TITLE
Adding Rust and API Template support

### PR DESCRIPTION
**Pull request by mor1**

### CODE REVIEW

File: .github/actions/create_api_docker_images/action.yaml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+  image_tag:
+    required: true
+    description: "Image tag to use for this chart"
+
+runs:
+  using: "composite"
+  steps:
+    - run: |
+        export DOCKER_BUILDKIT=1
+        image_base="$DOCKER_REGISTRY/$(echo -n $GITHUB_REPOSITORY | sed 's|kvasira|kvasir-docker-registry|')"
```
Comment (chris-ikva) [inline .github/actions/create_api_docker_images/action.yaml:22]: as a public repo we should not be exposing the name of our registry

**Review by chris-ikva — CHANGES_REQUESTED** [review]

File: .github/workflows/rust.yml [hunk 2]
```diff
@@ -0,0 +1,77 @@
...
+        with:
+          profile: minimal
+          toolchain: stable
+          override: true
+      - uses: actions-rs/cargo@v1
+        with:
+          command: test
+
+  fmt:
+    name: Rustfmt
```
Comment (chris-ikva) [inline .github/workflows/rust.yml:45]: I am ok with a pipeline validating that the code is well formatted but it shouldn't change the format - does this just validate?

Comment (mor1) [reply .github/workflows/rust.yml:45]: oh -- good point, will fix

Comment (mor1) [reply .github/workflows/rust.yml:45]: (actually, yes it should just run it in `check` mode exiting with non-zero status if not correctly formatted)

**Pull request merged**